Index: src/main/java/controller/studentActions/DisplayAllGradesAction.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package controller.studentActions;\r\n\r\nimport controller.Action;\r\n\r\npublic class DisplayAllGradesAction implements Action {\r\n    @Override\r\n    public void executeQuery() {\r\n\r\n    }\r\n\r\n    @Override\r\n    public String getlabel() {\r\n        return \"Display all grades\";\r\n    }\r\n}\r\n
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/controller/studentActions/DisplayAllGradesAction.java b/src/main/java/controller/studentActions/DisplayAllGradesAction.java
--- a/src/main/java/controller/studentActions/DisplayAllGradesAction.java	(revision 029027e04def22cdcda688b2d5610d9ddfbcec5d)
+++ b/src/main/java/controller/studentActions/DisplayAllGradesAction.java	(date 1640102569013)
@@ -1,11 +1,55 @@
 package controller.studentActions;
 
 import controller.Action;
+import controller.SQLController;
+import repository.UserDAO;
+import view.MainView;
+
+import java.sql.*;
 
 public class DisplayAllGradesAction implements Action {
     @Override
     public void executeQuery() {
+        UserDAO userDAO=new UserDAO();
+        String studentid="null";
+        try {
+            studentid=String.valueOf(userDAO.getId(MainView.getLogin()));
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
+        String query="SELECT subject_name ,grade FROM dziennik.grades, dziennik.subjects where dziennik.grades.subject_id=dziennik.subjects.subject_id and\n" +
+                "student_id="+studentid+" order by subject_name ; ";
+        Connection connection = null;
+
+        try {
+            connection = DriverManager.getConnection(SQLController.URL, SQLController.USERNAME,SQLController.PASSWORD);
+            Statement statement = connection.createStatement();
+            ResultSet resultSet = statement.executeQuery(query);
+
+            String oldsubject="";
+            while(resultSet.next()){
+                String subject=resultSet.getString("subject_name");
+                String grade=resultSet.getString("grade");
+                if(subject.equals(oldsubject)){
+                    System.out.print(","+grade);
+
+                }
+                else{
+                    System.out.println();
+                    System.out.print(subject+":"+grade);
+
+
+
+                }
+
+                oldsubject=subject;
 
+            }
+            System.out.println();
+
+        } catch (SQLException throwables) {
+            throwables.printStackTrace();
+        }
     }
 
     @Override
